### ISSUE TITLE
TabPage Click event raised when switching tabs via keyboard

### ISSUE THREAD

**Issue by LaraSQP** [picoe/Eto#1593]
Essentially, selection of a tab whichever way raises the `Click` event.

Even when adding the `TabPage` to a `TabControl` raises the `Click` event.

It is likely this is also the reason why `Click` provides a `EventArgs` rather than a `MouseEventArgs` parameter.

**Comment by cwensley** [comment]
Yes, this might have been done on purpose as an event to tell when that TabPage has been selected.  It might not be named very well.  I'll investigate its behaviour and may rename it to something else, obsoleting the `Click` event if this is the case.

**Comment by LaraSQP** [comment]
Please consider also implementing a `Click` event with `MouseEventArgs`.

This would be useful to display a context menu or close a tab on right-click.

**Comment by cwensley** [comment]
It will be considered, but it may be _very_ difficult to implement right on all platforms.  Something like that will probably need to be done during a MouseDown/MouseUp event vs. some sort of Click event.

**Comment by LaraSQP** [comment]
`MouseDoubleClick` has `MouseEventArgs` already implemented.

That is how I have to do it now, `ctrl + right-double-click` for tab closing, which is kind of weird admittedly. I figure I would be doing some similar thing for context menus when I get there.

**Comment by cwensley** [comment]
Yes, correct.  However there's no way of knowing if the mouse is over the tab vs. the content right now.

If you're looking for a tab control with closing, have you looked at the DocumentControl?

**Comment by LaraSQP** [comment]
Yes, I have been playing with the `DocumentControl`. It might just do.
